pkg/deploy/controller/deployment/controller.go

```diff
@@ -289,6 +290,14 @@ func (c *DeploymentController) makeDeployerPod(deployment *kapi.ReplicationContr
 
 	gracePeriod := int64(10)
 
+	// In case the deployment config strategy does not set the resource requests,
+	// we default them.
+	if deploymentConfig.Spec.Strategy.Resources.Limits == nil {
+		deploymentConfig.Spec.Strategy.Resources.Limits = kapi.ResourceList{}
+		deploymentConfig.Spec.Strategy.Resources.Limits[kapi.ResourceCPU] = resource.MustParse("100m")
+		deploymentConfig.Spec.Strategy.Resources.Limits[kapi.ResourceMemory] = resource.MustParse("256Mi")
```

Comment:
@smarterclayton for custom deployments you can always override this in the strategy.

If I set requests I get this:

```
pods "history-limit-4-hook-pre" is forbidden: Failed quota: quota-terminating: must specify limits.cpu,limits.memory,requests.cpu,requests.memory (ignore), deployment test/history-limit-4 will continue
```